forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt

```diff
@@ -121,18 +129,23 @@ interface DeviceAuthenticator {
      * sign the challenge sent from the server and generate signed JWT
      * @param userKey User Information
      * @param challenge challenge received from server
+     * @param customClaims A map of custom claims to be added to the jws payload
      */
     fun sign(context: Context,
              userKey: UserKey,
              privateKey: PrivateKey,
              signature: Signature?,
              challenge: String,
-             expiration: Date): String {
-
+             expiration: Date,
+             customClaims: Map<String, Any> = emptyMap()): String {
+        val customClaimsSetBuilder = JWTClaimsSet.Builder()
+        customClaims.forEach { (key, value) ->
+            customClaimsSetBuilder.claim(key, value)
+        }
         val signedJWT =
             SignedJWT(JWSHeader.Builder(parse(getAlgorithm()))
                 .keyID(userKey.kid).build(),
-                JWTClaimsSet.Builder().subject(userKey.userId)
+                JWTClaimsSet.Builder(customClaimsSetBuilder.build()).subject(userKey.userId)
```

Comment:
Should we set the custom claim after the default? so that it can override the existing one, just in case the developer intends to override the existing claim.